src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -65,7 +65,7 @@ internal class CircuitRegistry
             };
         }
 
-        internal ConcurrentDictionary<string, CircuitHost> ConnectedCircuits { get; }
+        internal ConcurrentDictionary<CircuitSecret, CircuitHost> ConnectedCircuits { get; }
```

Comment:
The one thing that's a little tricky about all of this BTW is that `MemoryCache` is weakly typed for keys. You'll always see `CircuitSecret.Secret` passed in there, but it's something to review carefully.

Reply:
If we want, we could standardize in the orther direction, but that would box. Either way, if we do something inconsistent it will have bugs. Boo weakly-typed-ness 👎